public/docs/ts/latest/cookbook/a11y.jade

```diff
@@ -0,0 +1,1087 @@
+  a button, regardless of the original design of the HTML element. More information on this later.
+  
+  Setting the `tabindex` to `0` inserts the element in the default flow of`keyboard navigation` focus. This means that our
+  element also becomes accessible via the keyboard!
+
+.l-sub-section
+  :marked
+    **DO NOT** use a `tabindex` value of `1` or greater as this will change the default keyboard navigation.
+
+:marked
+  ### Internal focus management for components
+
+  Programmatically setting focus within an app can introduce accessibility issues by interrupting
+  or enforcing a particular flow. If you do set focus for the user, you must ensure that
+  you do so in an accessible way.
+  
+  We'll illustrate how to do this by creating a `button` that shows an `alert`, then sets focus on the `alert`
+  and then allows the user to close the error message with a `close button`.
+
++makeTabs('cb-a11y/ts/app/managing-focus/a11y-error-demo.component.html,cb-a11y/ts/app/managing-focus/a11y-error-demo.component.ts',
+null, 'a11y-error-demo.component.html,a11y-error-demo.component.ts')
+
+:marked
+  We are setting focus on an `alert` that starts out hidden. As `hidden` and `disabled` elements cannot accept focus,
+  we first need this element to become visible again. To give the browser time to apply this change we set our focus using
+  a `timeout` function.
+  
+  Also note that we are using the `local template variable` to easily set focus right inside our
+  template code.
+
+  We also use `ARIA` to apply the role of `alert` and manage the `aria-hidden` property.
+
+:marked
+  ### Section summary
+  
+  In this section we saw how important `keyboard focus` is to make sure that many of our users are able to 
+  navigate our web pages.
+  
+  We looked at displaying the current `keyboard focus` and how to build our component templates with a 
+  natural flow of focus in mind.
+  
+  Finally we looked at what we can do to make sure that our own `components` can accept `focus` and how to 
+  programmatically manage `focus` in our components.
+  
+  Keep reading to further explore accessibility in Angular&nbsp;2 or [go back to the table of contents](#toc)
+
+.l-main-section
+<a id="component-roles"></a>
+:marked
+  ## Roles for custom component widgets
+
+  When creating a custom component widget, we can extend or change the behavior of an HTML element. To
+   ensure accessibility, we must map our new component's behavior to an appropriate role in the [Accessibility Tree](https://www.w3.org/WAI/PF/aria-implementation/#intro_treetypes).
+
+:marked
+  ### How to apply an ARIA Role in Angular&nbsp;2
+  
+  We give an `ARIA Role` to an element by setting the value of the `role` attribute.
+
+  When we write this directly in our HTML
+  it is as simple as:
+
+code-example(language="html" escape="html" format="linenums").
+  <h2 role="alert">I am an alert.</h2>
```

Comment:
I don't see it doing that here. It is not changing the semantic meaning of the element itself buit indicating that it is important information. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role